### PR TITLE
fix: bucket encryption is enabled if SSEAlgorithm is specified

### PR DESCRIPTION
**Pull request by int-tt**
Resolves: https://github.com/aquasecurity/defsec/issues/1067

### CODE REVIEW

File: internal/adapters/cloud/aws/s3/s3_test.go [hunk 1]
```diff
@@ -338,7 +338,6 @@ func bootstrapBucketEncryption(t *testing.T, api *s3api.Client, ctx context.Cont
 					ApplyServerSideEncryptionByDefault: &s3types.ServerSideEncryptionByDefault{
 						SSEAlgorithm: s3types.ServerSideEncryptionAes256,
 					},
-					BucketKeyEnabled: true,
```
Comment (simar7) [inline internal/adapters/cloud/aws/s3/s3_test.go:341]: I would keep this in, as it is another way for a bucket to be encrypted. As for the test, if you implement the suggestion I mentioned above, the test should be green for the cases when either `BucketKeyEnabled` and / or `ApplyServerSideEncryptionByDefault` is set.

Ideally they should be two different cases but I think we can leave that for now if you wish.

**Review by simar7 — CHANGES_REQUESTED** [review]
thanks looks good, I left some feedback in the comments.

**Comment by int-tt** [comment]
Thank you for feedback!
I fixed it.

**Comment by simar7** [comment]
> Thank you for feedback!
> I fixed it.

Looks like there's a few checks that are red, could you address those?

**Review by simar7 — APPROVED** [review]
thanks!

**Pull request merged**